### DISCUSSION TITLE
Illusioners

### DISCUSSION THREAD

**Discussion by Decme** [PurpurMC/Purpur#782] [__Dismissed__] [answered]
Is it possibile to give them a spawn chance?

example:
> zombie_horse
>   spawn-chance: 0.0

request:
> illusioner
>   spawn-chance: ??

**Comment by granny** [comment, marked as answer]
You can set custom spawns using a datapack. Use slicedlime's example datapack as a base to work off of https://github.com/slicedlime/examples/

**Comment by Decme** [comment]
It would be so much better to have built into Purpur; the problem with doing it via a datapack, is they over-write the entire biome data. So even with minor updates or across snapshots, it could result in inconsistencies or the datapack breaking altogether.

I'll give an example: Using the mob heads datapack from 1.16 - drowned naturally drop iron. 1.17 comes around, and suddenly drowned are supposed to drop copper. Except they still drop iron, because the 1.16 mob head datapack over-writes the entire mob droptable instead if simply adding to the droptable.

Likewise, for a custom mob table datapack, if a feature seed of a biome changes (say for example with 1.18.2), suddenly you're stuck with an outdated datapack that, while it works, it does not provide an accurate experience for whatever the current version may be, unless it was designed for the exact version you've created it for.

Dumbing it down, it's the difference between these:
Let X = vanilla variable (1 for example)

Plugin or build in Purpur feature would do this:
Let X = X + 1

Datapack would do this:
Let X = 2
Which is fine, until X in vanilla no longer = 1

In the meantime, I've created a 1.18 datapack for illusioners for whoever would like to use it:
https://www.file.io/download/J7wcUTybXjxB
